packages/apps/ability-uniswap-swap/src/lib/vincent-ability.ts

```diff
@@ -119,19 +94,35 @@ export const vincentAbility = createVincentAbility({
       });
     }
 
+    // Validate the provided route to ensure it's legitimate and uses correct parameters
+    const routeValidation = validateUniswapRoute({
+      route,
+      chainId: chainIdForUniswap,
+      tokenInAddress,
+      tokenInAmount: ethers.utils.parseUnits(tokenInAmount.toString(), tokenInDecimals).toString(),
```

Comment:
If we have time for it, can we drop decimals from the input params and use the tokens as base units instead of expecting an array with decimal indicator?
That would make this and validation simpler (just a \d+ string, no point, no symbols (-)) and avoids issues when consumer inputs a wrong decimals value

Using decimals is pretty much a UI thing for humans, there is almost no reason to use it in pure logic stuff like this

Reply:
It looks like there's only [one place](https://github.com/LIT-Protocol/Vincent/blob/4a642ff6ab4ed7f4eb52415720e782348065756a/packages/apps/ability-uniswap-swap/src/lib/ability-helpers/get-uniswap-quote.ts#L44) where we require the number of decimals for `tokenIn` and it's a required param for the Uniswap SDK that they presumably use for formatting the token values when fetching routes